### DISCUSSION TITLE
Running a signet mempool with signet styling

### DISCUSSION THREAD

**Discussion by 1ma** [mempool/mempool#5155] [Q&A] [answered]
I've been running a signet mempool for some time and it works fine, if I recall correctly it was as easy as setting `MEMPOOL.NETWORK` to `signet` in the `backend/mempool-config.json` file.

However I haven't had success running this mempool instance with the "signet styling", meaning the mempool looks like it is mainnet, the URLs do not start with `/signet`, there is no alert banner etc.

I wonder what additional setup needs to be done to get this working (likely in the frontend part?). I couldn't figure it out with the available documentation.

Running 3.0.0-dev8 on both backend and frontend.

**Comment by softsimon** [comment]
This is because the frontend only use the Signet UI if the URL is /signet, we solve this with Nginx routing in production.

We would need to add a new frontend config where we can specify the default network so that / can be the network of choice.

**Comment by 1ma** [comment]
Yeah I noticed that e.g. https://sigpool.bitcoinbarcelona.xyz/signet/block/0000022812b0feb768e847a785d54a0e3696919a4df8ac79f8bb2acba982b65a renders fine, except that it doesn't find the data to populate the page unless you then remove the /signet prefix from the URL.

It feels like a strategic URL rewrite somewhere in the `nginx-mempool.conf` file should do the trick, but I haven't been able to make sense of the production nginx config.

Edit: Just remembered now that there is indeed a `SIGNET_ENABLED` config in `frontend/mempool-frontend-config.json`, and I built the site with it on.

**Comment by softsimon** [comment, marked as answer]
Running with this Pull request should solve your problems. It will treat Signet as the main net.

https://github.com/mempool/mempool/pull/5156

**Reply by 1ma** [reply]
Sweet. Will give it a try and report back.

**Comment by softsimon** [comment]
Ok @1ma please try the PR now again. All issues should be resolved. Please test and reply back if everything works fine. Note that I updated the config name to ROOT_NETWORK

**Reply by 1ma** [reply]
Indeed https://github.com/mempool/mempool/pull/5156/commits/c00d2f3763e7949fa2ab931325cadc94e5351f5b has resolved all the issues I found last week. Thanks, man.

**Reply by 1ma** [reply]
Will `ROOT_NETWORK` work in the `mempool/frontend` Docker image? Given that it seems a "compile time" knob :thinking:
